CHANGELOG.md

```diff
@@ -11,6 +11,7 @@ Changelog for NeoFS Node
 - Container size estimation contract writing (#2819)
 - Custom contract deployment with custom zone via neofs-adm (#2827)
 - Errors in neofs-adm morph dump-names output (#2831)
+- Endless GC cycle if an object is stored on an unexpected shard (#2821)
```

Comment:
How can it be stored there? Especially if metabase was resynchronized?

Reply:
> How can it be stored there?

i don't know, shard list may change, disk may answer with error. so many cases may lead to the attempt to write an object to another shard. current code does not think that is bad, it just tries the next one

> Especially if metabase was resynchronized?'

what does it change? there is always a shard that is the most "good" for an object, if it stores it, the object is never moved. is _that shard's_ peapod stores info about, it will always refill the same (this shard's one) metabase, even if this shard is not "the best one" for this object